Geometry/TrackerGeometryBuilder/src/TrackerGeomBuilderFromGeometricDet.cc

```diff
@@ -222,36 +224,66 @@ void TrackerGeomBuilderFromGeometricDet::buildSilicon(std::vector<const Geometri
 
 
 void TrackerGeomBuilderFromGeometricDet::buildGeomDet(TrackerGeometry* tracker){
+
   PlaneBuilderForGluedDet gluedplaneBuilder;
-  auto  const & gdu= tracker->detUnits();
-  auto  const & gduId = tracker->detUnitIds();
+  auto const & gdu = tracker->detUnits();
+  auto const & gduId = tracker->detUnitIds();
+  auto const & gduType = tracker->detTypes();
+
+  if(gdu.size() != gduId.size() || gdu.size() != gduType.size())
+    edm::LogError("BuildingGeomDetUnits") << "detUnits, detUnitIds and detTypes vectors in TrackerGeometry should be the same size!";
```

Comment:
The gduType has been introduced at the beginning of my study, and now it is not used anymore in the code. It is not supposed to be that the length of the three vectors are the same, since more than one det can have the same type. I am going to remove the object and the Error message asap.